R/cluster-call.R

```diff
@@ -18,16 +20,22 @@
 #' cluster_call(cl, runif(1))
 #'
 #' # use ptype to simplify
-#' cluster_call(cl, runif(1), ptype = double())
+#' cluster_call(cl, runif(1), simplify = TRUE)
 #'
 #' # use cluster_send() to ignore results
 #' cluster_send(cl, x <- runif(1))
-#' cluster_call(cl, x, ptype = double())
-cluster_call <- function(cluster, code, ptype = list()) {
+#' cluster_call(cl, x, simplify = TRUE)
+cluster_call <- function(cluster, code, simplify = FALSE, ptype = NULL) {
   stopifnot(is_cluster(cluster))
   code <- enexpr(code)
   to_rm <- attr(cluster, "cleaner")$reset()
 
+
+  if (!isTRUE(simplify) && !is.null(ptype)) {
```

Comment:
Maybe take `simplify = NULL` and use a different default depending on whether `ptype` is supplied? If not supplied, defaults to `FALSE`. If supplied, defaults to `TRUE`.

Do we want to generally support `simplify = NA` for consistency with purrr?

Reply:
I think we want to move this interface towards `list_simplify()` so I'd prefer some warning here. 

But you're right that I should I support `simplify = NA` in order to match the interface exactly.